pkg/render/elasticsearchclusterconfig.go

```diff
@@ -56,10 +56,19 @@ func NewElasticsearchClusterConfigFromConfigMap(configMap *corev1.ConfigMap) (*E
 		}
 	}
 
+	if configMap.Data["flowShards"] == "" {
+		return nil, fmt.Errorf("'flowShards' is not set")
+	} else {
+		if flowShards, err = strconv.Atoi(configMap.Data["flowShards"]); err != nil {
+			return nil, errors.Wrap(err, "'flowShards' must be an integer")
+		}
+	}
+
 	return &ElasticsearchClusterConfig{
 		clusterName: configMap.Data["clusterName"],
 		replicas:    replicas,
 		shards:      shards,
+		flowShards: flowShards,
```

Comment:
This bug wouldn't have happened if we'd called the main NewESCC() function here instead of replicating.  I think it's worth changing.